contracts/helpers/MultiBlockHarvester.sol

```diff
@@ -0,0 +1,244 @@
+    ) external onlyGuardian {
+        yieldBearingToDepositAddress[yieldBearingAsset] = newDepositAddress;
+    }
+
+    /**
+     * @notice Toggle the trusted status of an address
+     * @param trusted address to toggle the trusted status
+     */
+    function toggleTrusted(address trusted) external onlyGuardian {
+        isTrusted[trusted] = !isTrusted[trusted];
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                        TRUSTED FUNCTIONS
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Initiate a rebalance
+     * @param scale scale to apply to the rebalance amount
+     * @param yieldBearingAsset address of the yieldBearingAsset
+     */
+    function harvest(address yieldBearingAsset, uint256 scale, bytes calldata) external onlyTrusted {
+        if (scale > 1e9) revert InvalidParam();
+        YieldBearingParams memory yieldBearingInfo = yieldBearingData[yieldBearingAsset];
+        (uint8 increase, uint256 amount) = _computeRebalanceAmount(yieldBearingAsset, yieldBearingInfo);
+        amount = (amount * scale) / 1e9;
+        if (amount == 0) revert ZeroAmount();
+
+        try transmuter.updateOracle(yieldBearingAsset) {} catch {}
+        _rebalance(increase, yieldBearingAsset, yieldBearingInfo, amount);
+    }
+
+    /**
+     * @notice Finalize a rebalance
+     * @param yieldBearingAsset address of the yieldBearingAsset
+     */
+    function finalizeRebalance(address yieldBearingAsset, uint256 balance) external onlyTrusted {
+        try transmuter.updateOracle(yieldBearingAsset) {} catch {}
+        _adjustAllowance(yieldBearingAsset, address(transmuter), balance);
+        uint256 amountOut = transmuter.swapExactInput(
+            balance,
+            0,
+            yieldBearingAsset,
+            address(agToken),
+            address(this),
+            block.timestamp
+        );
+        address depositAddress = yieldBearingAsset == XEVT
+            ? yieldBearingToDepositAddress[yieldBearingAsset]
+            : address(0);
+        _checkSlippage(balance, amountOut, yieldBearingAsset, depositAddress, true);
+        agToken.burnSelf(amountOut, address(this));
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                        INTERNAL FUNCTIONS
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    function _rebalance(
+        uint8 typeAction,
+        address yieldBearingAsset,
+        YieldBearingParams memory yieldBearingInfo,
+        uint256 amount
+    ) internal {
+        if (amount > maxMintAmount) revert TooBigAmountIn();
+        agToken.mint(address(this), amount);
```

Comment:
could we maybe have a governance parameter where this amount cannot be greater than some threshold?
As a way to ensure that if sth goes off with the contract we don't harm ourselves too much

Reply:
That is maxMintAmount